src/main/scala/io/iohk/ethereum/blockchain/sync/Blacklist.scala

```diff
@@ -139,6 +221,93 @@ object Blacklist {
+      val description: String = Disconnect.reasonToString(Disconnect.Reasons.AlreadyConnected)
+    }
+    case object IncompatibleP2pProtocolVersion extends BlacklistReason {
+      val reasonType: BlacklistReasonType = IncompatibleP2pProtocolVersionType
+      val description: String = Disconnect.reasonToString(Disconnect.Reasons.IncompatibleP2pProtocolVersion)
+    }
+    case object NullNodeIdentityReceived extends BlacklistReason {
+      val reasonType: BlacklistReasonType = NullNodeIdentityReceivedType
+      val description: String = Disconnect.reasonToString(Disconnect.Reasons.NullNodeIdentityReceived)
+    }
+    case object ClientQuitting extends BlacklistReason {
+      val reasonType: BlacklistReasonType = ClientQuittingType
+      val description: String = Disconnect.reasonToString(Disconnect.Reasons.ClientQuitting)
+    }
+    case object UnexpectedIdentity extends BlacklistReason {
+      val reasonType: BlacklistReasonType = UnexpectedIdentityType
+      val description: String = Disconnect.reasonToString(Disconnect.Reasons.UnexpectedIdentity)
+    }
+    case object IdentityTheSame extends BlacklistReason {
+      val reasonType: BlacklistReasonType = IdentityTheSameType
+      val description: String = Disconnect.reasonToString(Disconnect.Reasons.IdentityTheSame)
+    }
+    case object TimeoutOnReceivingAMessage extends BlacklistReason {
+      val reasonType: BlacklistReasonType = TimeoutOnReceivingAMessageType
+      val description: String = Disconnect.reasonToString(Disconnect.Reasons.TimeoutOnReceivingAMessage)
+    }
+    case object OtherSubprotocolSpecificReason extends BlacklistReason {
+      val reasonType: BlacklistReasonType = OtherSubprotocolSpecificReasonType
+      val description: String = Disconnect.reasonToString(Disconnect.Reasons.Other)
+    }
+    case object EmptyStateNodeResponse extends BlacklistReason {
+      val reasonType: BlacklistReasonType = EmptyStateNodeResponseType
+      val description: String = "Empty state node response from peer"
+    }
+    case object WrongStateNodeResponse extends BlacklistReason {
+      val reasonType: BlacklistReasonType = WrongStateNodeResponseType
+      val description: String = "Fetched node state hash doesn't match requested one"
+    }
+    case object UnrequestedBodies extends BlacklistReason {
+      val reasonType: BlacklistReasonType = UnrequestedBodiesType
+      val description: String = "Received unrequested bodies"
+    }
+    final case class RegularSyncRequestFailed(error: String) extends BlacklistReason {
+      val reasonType: BlacklistReasonType = RegularSyncRequestFailedType
+      val description: String = s"Request failed with error: $error"
+    }
+    final case class BlockImportError(error: String) extends BlacklistReason {
+      val reasonType: BlacklistReasonType = BlockImportErrorType
+      val description: String = s"Block import error: $error"
+    }
+
+    private val allP2PReasons = List(
```

Comment:
It might be easier to use enumeratum to automatically generate that list. It is already in the dependencies but barely used. 
something like `findValues.filter(_.reasonType.isInstanceOf[P2PBlacklistGroup])`.
It would make that list easier to maintain and less error prone, but it would increase our dependency over enumeratum so I don't have a strong opinion on this one.

Reply:
this then also can be eliminated